File: 알고리즘/2/Excel Sheet Column Number.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int titleToNumber(string s) {
      string standard = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
      unordered_map<char, int> map;
      for (int i=0; i<standard.size(); i++){
        map[standard[i]] = i+1;
      }
      int result = 0;
      int a = 1;
      while (s.size()>0){
        if(s.size()==1){
          result+=(map[s[0]]*a);
          s="";
        } else{
            result += (map[s[s.size()-1]]*a);
            s = s.substr(0,s.size()-1); 
            a*=26;
        }  
      }
      return result;
      
    }
};

File: algorithms/2/Excel Sheet Column Number.py
class Solution:
    def titleToNumber(self, s: str) -> int:
        standard = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
        result = 0

        dict1 = {ele: idx+1 for idx, ele in enumerate(standard)}
        a = 1
        while len(s) > 0:
            if(len(s) == 1):
                result += (dict1[s[-1]]*a)
                s = ''
            else:
                result += (dict1[s[-1]]*a)
                s = s[:-1]
                a *= 26
        return result
